src/Shape_buildblock/GenerateImage.cxx

```diff
@@ -0,0 +1,307 @@
+  }
+  if (num_samples.z()<=0)
+  {
+    warning("number of samples to take in z-direction should be strictly positive\n");
+    return true;
+  }
+  if (num_samples.y()<=0)
+  {
+    warning("number of samples to take in y-direction should be strictly positive\n");
+    return true;
+  }
+  if (num_samples.x()<=0)
+  {
+    warning("number of samples to take in x-direction should be strictly positive\n");
+    return true;
+  }
+  return false;
+}
+
+//! \brief parses parameters
+/*! \warning Currently does not support interactive input, due to
+    the use of the 'next shape' keyword.
+*/
+GenerateImage::
+GenerateImage(const char * const par_filename)
+{
+  set_defaults();
+  initialise_keymap();
+  if (par_filename!=0)
+  {
+    if (parse(par_filename) == false)
+      exit(EXIT_FAILURE);
+  }
+  else
+    ask_parameters();
+
+#if 0
+  // doesn't work due to problem  Shape3D::parameter_info being ambiguous
+  for (vector<shared_ptr<Shape3D> >::const_iterator iter = shape_ptrs.begin();
+       iter != shape_ptrs.end();
+       ++iter)
+    {
+      std::cerr << (**iter).parameter_info() << '\n';
+    }
+#endif
+
+}
+
+
+Succeeded
+GenerateImage::
+compute()
+{
+#if 0
+  shared_ptr<DiscretisedDensity<3,float> > density_ptr(
+						       read_from_file<DiscretisedDensity<3,float> >(template_filename));
+  this->out_density_ptr =
+    density_ptr->clone();
+  out_density_ptr->fill(0);
+  VoxelsOnCartesianGrid<float> current_image =
+    dynamic_cast<VoxelsOnCartesianGrid<float>& >(*density_ptr);
+
+#else
+
+  if (image_duration>0.0)
+  {
+    std::vector<double> start_times(1, rel_start_time);
+    std::vector<double> durations(1, image_duration);
+    TimeFrameDefinitions frame_defs(start_times, durations);
+    exam_info_sptr->set_time_frame_definitions(frame_defs);
+  }
+  else
+  {
+    warning("image duration not set, so time frame definitions will not be initialised");
+  }
+  VoxelsOnCartesianGrid<float>
+          current_image(exam_info_sptr,
+                        IndexRange3D(0,output_image_size_z-1,
+                                     -(output_image_size_y/2),
+                                     -(output_image_size_y/2)+output_image_size_y-1,
+                                     -(output_image_size_x/2),
+                                     -(output_image_size_x/2)+output_image_size_x-1),
+                        CartesianCoordinate3D<float>(0,0,0),
+                        CartesianCoordinate3D<float>(output_voxel_size_z,
+                                                     output_voxel_size_y,
+                                                     output_voxel_size_x));
+  this->out_density_ptr.reset(current_image.clone());
+#endif
+  std::vector<float >::const_iterator value_iter = values.begin();
+  for (std::vector<shared_ptr<Shape3D> >::const_iterator iter = shape_ptrs.begin();
+          iter != shape_ptrs.end();
+  ++iter, ++value_iter)
+  {
+    std::cerr << "Next shape\n"; //(**iter).parameter_info() << '\n';
```

Comment:
probably comment this out. it isn't very useful. we'd have to call `info` instead (at verbosity 2) if you feel you want to keep it.